sweep/sweeper.go

```diff
@@ -949,6 +981,135 @@ func (s *UtxoSweeper) handlePendingSweepsReq(req *pendingSweepsReq) []*PendingIn
+	// simply just increase its fee rate. This will allow the input to be
+	// batched with others which also have a similar fee rate, creating a
+	// higher fee rate transaction that replaces the original input's
+	// sweeping transaction.
+	if pendingInput, ok := s.pendingInputs[*req.input.OutPoint()]; ok {
+		return s.handleRBFBump(req, pendingInput, bestHeight)
+	}
+
+	// If the input is unknown to the UtxoSweeper and we don't have
+	// information about the output needed to properly sweep it, then we'll
+	// return an error indicating so.
+	if req.input.SignDesc().Output == nil {
+		return nil, lnwallet.ErrNotMine
+	}
+
+	// Otherwise, we'll register the input within the UtxoSweeper, which
+	// will end up creating a child transaction that pays for its parent's
+	// fee rate.
+	return s.handleCPFPBump(req, bestHeight)
+}
+
+// handleRBFBump handles a bump fee request by simply updating the inputs fee
+// preference. Currently, no validation is done on the new fee preference to
+// ensure it will properly create a replacement transaction.
+//
+// TODO(wilmer):
+//   * Validate fee preference to ensure we'll create a valid replacement
+//     transaction to allow the new fee rate to propagate throughout the
+//     network.
+//   * Ensure we don't combine this input with any other unconfirmed inputs that
+//     did not exist in the original sweep transaction, resulting in an invalid
+//     replacement transaction.
+func (s *UtxoSweeper) handleRBFBump(req *bumpFeeReq, input *PendingInput,
+	bestHeight int32) (chan Result, error) {
+
+	log.Debugf("Updating fee preference for %v from %v to %v",
+		*input.OutPoint(), input.FeePreference, req.feePreference)
+
+	input.FeePreference = req.feePreference
+
+	// We'll reset the input's publish height to the current so that a new
+	// transaction can be created that replaces the transaction currently
+	// spending the input. We only do this for inputs that have been
+	// broadcast at least once to ensure we don't spend an input before its
+	// maturity height.
+	if input.PublishAttempts > 0 {
+		input.MinPublishHeight = bestHeight
+	}
+
+	if err := s.scheduleSweep(bestHeight); err != nil {
+		log.Errorf("Unable to schedule sweep: %v", err)
+	}
+
+	resultChan := make(chan Result, 1)
+	input.listeners = append(input.listeners, resultChan)
+
+	return resultChan, nil
+}
+
+// handleCPFPBump handles a bump fee request by creating a Child-Pays-For-Parent
+// (CPFP) transaction. The child transaction will spend the input we're
+// attempting to sweep, which should belong to the parent transaction.
+//
+// TODO(wilmer): Validate fee preference to ensure we're creating a child
+// transaction with a high enough fee rate to cover its parents.
+func (s *UtxoSweeper) handleCPFPBump(req *bumpFeeReq,
+	bestHeight int32) (chan Result, error) {
+
+	resultChan := make(chan Result, 1)
+	inputReq := &sweepInputMessage{
+		input:         req.input,
+		feePreference: req.feePreference,
```

Comment:
So it's the job of the caller to compute the proper fee rate that'll cause the entire package to be bumped by a satisfactory amount?

Reply:
Yes. I'm not a fan of this, but from our previous discussions offline it seemed like this was the approach to take for now. W should be able to compute this quite easily for full nodes, but things get a bit trickier with light clients.